torch/csrc/jit/codegen/cuda/lower_loops.cpp

```diff
@@ -90,100 +90,154 @@ void LoopNestGenerator::pushFront(kir::Expr* expr) {
 
-    // Look up the concrete ID in the parallel map, not in the loop
-    // map, which also maps non-CA axes.
-    auto concrete_id =
-        gpu_lower->caParallelMap().getConcreteMappedID(out_tv->axis(out_i));
-    loop_structure.push_back(concrete_id);
-  }
+    // Fill the entire loop structure by Looking at each axis
+    // individually in out's domain
+    for (size_t out_i = 0; out_i < out_tv->nDims(); out_i++) {
+      // Note: It is not safe to skip trivial reduction axes as they could be
+      // inlined with other tensor views. This happens in
+      // NVFuserTest.FusionBNRepro_CUDA as of this commit on norm_hack_2_rebased
+      // branch
 
-  auto loop_structure_it = loop_structure.begin();
-  auto for_loop_it = for_loops_.begin();
-  auto last_for_loop_matched = for_loops_.begin();
-
-  // Match the loop structure with the current for-loops. Reuse
-  // matching loops and close unmatched ones.
-  while (loop_structure_it != loop_structure.end() &&
-         for_loop_it != for_loops_.end()) {
-    auto lowered_out_id =
-        gpu_lower->lowerValue(*loop_structure_it)->as<kir::IterDomain>();
-    // Similar to the above, the parallel map is used rather than the
-    // loop map. Again, non-CA axes should not share loops, so the
-    // parallel map should be used.
-    if (gpu_lower->caParallelMap().areMapped(
-            lowered_out_id, (*for_loop_it)->iter_domain())) {
-      loop_structure_it++;
-      last_for_loop_matched = ++for_loop_it;
-    } else {
-      ++for_loop_it;
+      // Look up the concrete ID in the parallel map, not in the loop
+      // map, which also maps non-CA axes.
+      auto concrete_id =
+          gpu_lower->caParallelMap().getConcreteMappedID(out_tv->axis(out_i));
+      loop_structure.push_back(concrete_id);
     }
-  }
 
-  auto n_loops_to_close =
-      std::distance(last_for_loop_matched, for_loops_.end());
+    auto loop_structure_it = loop_structure.begin();
+    auto for_loop_it = for_loops_.begin();
+    auto last_for_loop_matched = for_loops_.begin();
+
+    // Match the loop structure with the current for-loops. Reuse
+    // matching loops and close unmatched ones.
+    while (loop_structure_it != loop_structure.end() &&
+           for_loop_it != for_loops_.end()) {
+      auto lowered_out_id =
+          gpu_lower->lowerValue(*loop_structure_it)->as<kir::IterDomain>();
+      // Similar to the above, the parallel map is used rather than the
+      // loop map. Again, non-CA axes should not share loops, so the
+      // parallel map should be used.
+      if (gpu_lower->caParallelMap().areMapped(
+              lowered_out_id, (*for_loop_it)->iter_domain())) {
+        loop_structure_it++;
+        last_for_loop_matched = ++for_loop_it;
+      } else {
+        ++for_loop_it;
+      }
+    }
 
-  for (int64_t i = 0; i < n_loops_to_close; i++) {
-    closeFor();
-  }
+    auto n_loops_to_close =
+        std::distance(last_for_loop_matched, for_loops_.end());
```

Comment:
Just to confirm, nothing has changed so far in this function except it's in a loop traversing the given expression list. Let me know if I'm missing something.